transport/src/main/java/io/netty/channel/FixedRecvByteBufAllocator.java

```diff
@@ -38,6 +41,22 @@ public int guess() {
         }
     }
 
+    /**
+     * Creates a new {@link FixedRecvByteBufAllocator} that will continue reading even when the read
+     * {@link io.netty.channel.socket.DatagramPacket} was smaller then the given {@code bufferSize}.
+     *
+     * @param bufferSize the buffer size that is used for {@link io.netty.channel.socket.DatagramPacket}s.
+     * @return the allocator.
+     */
+    public static FixedRecvByteBufAllocator newDatagramRecvByteBufAllocator(int bufferSize) {
```

Comment:
I think this could be shortened to `forDatagrams`.

Do we normally use other implementations, like Adaptive*, for TCP? Otherwise a case could be made to have a companion method for streams.

Reply:
`Adaptive*` may also make sense in some cases... but usually we use the `FixedRecvByteBufAllocator`